File: app/syncthing/service_config.py
from dataclasses import dataclass

DEFAULT_TIMEOUT = 10

@dataclass
class ServiceConfig:
    api_key: str = ''
    host: str = 'localhost'
    port: int = 8384
    timeout: int = DEFAULT_TIMEOUT
    is_https: bool = False
    ssl_cert_file: str | None = None

File: app/src/syncthing/ServiceConfig.ts
// in seconds
export const DEFAULT_TIMEOUT = 100;

export class ServiceConfig {
  constructor(
    public apiKey: string,
    public host: string = 'localhost',
    public port: number = 8384,
    public timeout: number = DEFAULT_TIMEOUT,
    public isHttps: boolean = false,
    public sslCertFile: string | undefined = undefined,
  ) {
  }
}
